code/engine/xrCore/_math.cpp

```diff
@@ -201,90 +199,10 @@ void _initialize_cpu() {
     Didentity.identity();  // Identity matrix
     pvInitializeStatics(); // Lookup table for compressed normals
     FPU::initialize();
-    _initialize_cpu_thread();
-
-    g_initialize_cpu_called = true;
-}
-
-// per-thread initialization
-#include <xmmintrin.h>
-#define _MM_DENORMALS_ZERO_MASK 0x0040
-#define _MM_DENORMALS_ZERO_ON 0x0040
-#define _MM_FLUSH_ZERO_MASK 0x8000
-#define _MM_FLUSH_ZERO_ON 0x8000
-#define _MM_SET_FLUSH_ZERO_MODE(mode) _mm_setcsr((_mm_getcsr() & ~_MM_FLUSH_ZERO_MASK) | (mode))
-#define _MM_SET_DENORMALS_ZERO_MODE(mode) \
-    _mm_setcsr((_mm_getcsr() & ~_MM_DENORMALS_ZERO_MASK) | (mode))
-static BOOL _denormals_are_zero_supported = TRUE;
-extern void __cdecl _terminate();
-void debug_on_thread_spawn();
-
-void _initialize_cpu_thread() {
-    debug_on_thread_spawn();
 #ifndef XRCORE_STATIC
-    // fpu & sse
-    FPU::m24r();
+	// fpu & sse
+	FPU::m24r();
 #endif // XRCORE_STATIC
-    if (CPU::ID.hasFeature(CpuFeature::Sse)) {
-        //_mm_setcsr ( _mm_getcsr() | (_MM_FLUSH_ZERO_ON+_MM_DENORMALS_ZERO_ON) );
-        _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
-        if (_denormals_are_zero_supported) {
-            __try {
-                _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);
-            } __except (EXCEPTION_EXECUTE_HANDLER) {
-                _denormals_are_zero_supported = FALSE;
-            }
-        }
-    }
-}
-
-// threading API
-#pragma pack(push, 8)
-struct THREAD_NAME {
-    DWORD dwType;
-    LPCSTR szName;
-    DWORD dwThreadID;
-    DWORD dwFlags;
-};
-void thread_name(const char* name) {
```

Comment:
It is a useful function, but her implementation is a little bit outdated. Let's use SetThreadDescription on win 10 and nothing on the other win versions.